curator-client/src/main/java/org/apache/curator/ConnectionState.java

```diff
@@ -213,16 +213,14 @@ private synchronized void checkTimeouts() throws Exception
      * Return the current session id
      */
     public long getSessionId() {
-        long sessionId = -1;
-        if (isConnected()) {
-            try {
-                ZooKeeper zk = zooKeeper.getZooKeeper();
-                if (zk != null) {
-                    sessionId = zk.getSessionId();
-                }
-            } catch (Exception e) {
-                // Ignore the exception
+        long sessionId = 0;
```

Comment:
@Randgalt, here we're calling the HandleHolder.getZookeeper() directly, so it won't call closeAndReset(), please correct me if I missed anything.